gce.py

```diff
@@ -215,8 +215,12 @@ def node_to_dict(self, inst):
                 md[entry['key']] = entry['value']
 
         net = inst.extra['networkInterfaces'][0]['network'].split('/')[-1]
+        # If there is no public IP listed, set the public_ip field to a null.
         if inst.public_ips.__len__() == 0:
           inst.public_ips=[None]
+
+        # Use private IPs if we're told to, or we've got no choice.
+        if os.environ.has_key("ANSIBLE_USE_PRIVATE_SSH") or inst.public_ips==[None]:
```

Comment:
The default from ansible is to use public_ips... If we break their default, there is little chance that they'll ever accept this pull upstream. The use of the environment variable was to ensure the default for ansible remained unchanged.